src/redux/Books/Books.js

```diff
@@ -0,0 +1,30 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const bookSlice = createSlice({
+  name: 'add_remove_book',
+  initialState: {
+    books: [],
```

Comment:
- [ ] Great job so far, we should defined action types and use [ducks pattern](https://github.com/erikras/ducks-modular-redux), to define them, kindly add that.

Reply:
Okay, thanks. I'II walk through it.

Reply:
@Wes-Isaac
, thanks for the comment but I have a question. my question is that I used the redux toolkit so that action types are created automatically and I think the ducks pattern used the redux core. so should I create action types manually?

Reply:
@degisew, you can still define action with redux toolkit, you can checkout this [link](https://redux-toolkit.js.org/api/createAction)

Reply:
@degisew  you can use the actionCreator from redux toolkit

```
import { createAction } from '@reduxjs/toolkit'
const increment = createAction('counter/increment')
```
something like this add use this.

Reply:
Okay. Thanks for your help.